src/Stateless/StateMachine.Async.cs

```diff
@@ -144,7 +144,7 @@ public Task FireAsync<TArg0>(TriggerWithParameters<TArg0> trigger, TArg0 arg0)
         /// is fired.
         /// </summary>
         /// <param name="unhandledTriggerAction">An asynchronous action to call when an unhandled trigger is fired.</param>
-        public void OnUnhandledTriggerAsync(Func<TState, TTrigger, Task> unhandledTriggerAction)
+        public void OnUnhandledTriggerAsync(Func<TState, TTrigger, ICollection<string>, Task> unhandledTriggerAction)
```

Comment:
The old `OnUnhandledTriggerAsync()` overload is missing here (for the non-async version we ended up with a new overload alongside the old one)